src/main/java/io/workflow/bpmnsimulator/service/ProcessService.java

```diff
@@ -0,0 +1,34 @@
+package io.workflow.bpmnsimulator.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.camunda.bpm.engine.ProcessEngineServices;
+import org.camunda.bpm.engine.RuntimeService;
+import org.camunda.bpm.engine.repository.ProcessDefinition;
+import org.camunda.bpm.engine.runtime.ActivityInstance;
+import org.camunda.bpm.engine.runtime.ProcessInstance;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.Nonnull;
+import java.util.Optional;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ProcessService {
+
+    private final RuntimeService runtimeService;
+
+    @Nonnull
+    public ProcessInstance startProcessInstance(@Nonnull final ProcessDefinition processDefinition) {
+        final ProcessInstance processInstance = runtimeService.startProcessInstanceById(processDefinition.getId());
+        log.debug("A new process instance: [{}] started with process definition: [{}]", processInstance, processDefinition);
+
+        return processInstance;
+    }
+
+    public Optional<ActivityInstance> getProcessInstance(@Nonnull final String processInstanceId) {
```

Comment:
if you want to get process instance you must be used ProcessInstanceQuery and filter by processInstanceId.
if you get Activity Instance, activity is all of item in processInstance like task, sub process and ... . I think that must be write to other class like ActivityService